extensions/logs/suite.go

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2021 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logs
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+var logContextTimeoutFlag = flag.String("log-context-timeout", "15s",
+	"log context timeout")
+var logDirFlag = flag.String("log-dir", "logs",
+	"container logs directory")
+var ignoreNamespaceFlag = flag.String("ignore-k8s-ns", "kube-system,local-path-storage",
+	"comma separated list of ignored kubernetes namespaces")
+
+type Suite struct {
+	suite.Suite
+
+	testStartTime time.Time
+	ctxTimeout    time.Duration
+	kubeClient    kubernetes.Interface
+}
+
+func (s *Suite) SetupSuite() {
+	var err error
+
+	s.ctxTimeout, err = time.ParseDuration(*logContextTimeoutFlag)
+	require.NoError(s.T(), err)
+	require.Greater(s.T(), s.ctxTimeout, 0)
+
+	s.kubeClient, err = newKubeClient()
+	require.NoError(s.T(), err)
+}
+
+func (s *Suite) SetupTest() {
+	s.testStartTime = time.Now()
+}
+
+func (s *Suite) AfterTest(suiteName, testName string) {
+	logPath := fmt.Sprintf("%s/%s/%s", *logDirFlag, suiteName, testName)
```

Comment:
Why do we need to parse flag after each test?

Reply:
According to documentation, flags parsed on `flag.Parse()` call, thus this is simple access to already parsed value. I am sure we should avoid to create excess entities.

Reply:
Indeed, I looked at this quickly and missed this :)